lib/Hook.js

```diff
@@ -8,30 +8,30 @@ exports.trigger = function () {
   }
 };
 
-exports.wait = function () {
-  var args = Array.prototype.slice.apply(arguments);
-  var self = args.shift();
-  var hook = args.shift();
-  var next = args.shift();
+exports.wait = function (self, hook, next, opts) {
+  if (typeof hook !== "function") {
+    return next();
+  }
 
-  args.push(next);
```

Comment:
This was a bit complicated and it wasn't clear what it's actually doing (there's another optional `opts` parameter which gets passed to the hook).
Rewritten to make it more explicit.